### PR TITLE
fix: [ANDROAPP-2738] fixes deletion of a Tei or an Enrollement

### PR DESCRIPTION
**Pull request by ferdyrod**
**JIRA ISSUE**: [ANDROAPP-2738](https://jira.dhis2.org/browse/ANDROAPP-2738)

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java [hunk 1]
```diff
@@ -589,6 +598,22 @@ else if (d2.trackedEntityModule().trackedEntityAttributes().byUid().eq(tetAttrib
...
+                if (program != null) {
+                    Enrollment enrollment = d2.enrollmentModule().enrollments().byTrackedEntityInstance().eq(tei.uid()).byProgram().eq(program.uid()).one().blockingGet();
+                    if (enrollment.program().equals(program.uid()) && enrollment.deleted()) {
+                        iterator.remove();
+                    }
+                }
+            }
+        }
+        return teis;
+    }
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: to me this should go outside repository, it's not its responsability

Comment (JaimeToca) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: thoughts? @ferdyrod

Comment (ferdyrod) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: Yes, it should. this whole repository needs a lot of rework.

Comment (JaimeToca) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: :+1:  great ! do you think this small piece can be extracted (only filterDeletedEnrollment) or has other considerations??

Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: I'm going to extract it to an extension function

**Pull request merged**